package/Runtime/Plugins/iOS/NimbusManager.swift

```diff
@@ -12,19 +12,21 @@ import NimbusKit
 
 @objc public class NimbusManager: NSObject {
     
-    @objc public static let shared = NimbusManager()
+    private static var managerDictionary: [Int: NimbusManager] = [:]
```

Comment:
Does this dictionary remove ads that have been destroyed or have completed their lifecycle @brugg ?  If not this will leak memory as the number of items in the dictionary grows.  Looking at the code, unless I missed it see you adding to this dictionary by never removing from it.

Reply:
True. Good catch.

Reply:
https://github.com/timehop/nimbus-unity/pull/21/commits/efb15c0c9660c5175267221c46bf9ff3037fad8f